In [1]:
import pandas as pd
import numpy as np
import multiprocessing as mp
from time import time

In [2]:
time1 = time()
random=np.random.normal(10,0.1,(20000,20000))
df=pd.DataFrame(random,index=np.arange(20000), columns=np.arange(20000))
print(np.mean(df[::100],axis=0))

print('\nPool Parallelism runs %0.3f seconds using numpy\n' % ((time()-time1)))


0        10.003895
1         9.996079
2         9.988127
3         9.995153
4        10.007450
           ...    
19995     9.994788
19996     9.995534
19997    10.002427
19998     9.995828
19999    10.002994
Length: 20000, dtype: float64

Pool Parallelism runs 16.948 seconds using numpy



In [3]:
import dask.array as da


In [4]:
time1 = time()
daskrandom=da.random.normal(10,0.1,size=(20000,20000),chunks=(1000, 1000))
daskmean = daskrandom[::100].mean(axis=0)
print(daskmean.compute())
print('\nPool Parallelism runs %0.3f seconds using Dask\n' % ((time()-time1)))

[ 9.99920079 10.01479739 10.00941168 ...  9.98747687  9.99725238
 10.01142332]

Pool Parallelism runs 6.138 seconds using Dask



## Performance Comparision

To mesasure the peroformance we calculate the time taken by the numpy array and the dask array, The numpy array took 16.948 sec to finish the task wheraas the dask array only took 6.138 sec to complete the task. In comparision the dask array is faster than the numpy array.
